src/bp/core/routers/telemetry.ts

```diff
@@ -0,0 +1,64 @@
+import * as sdk from 'botpress/sdk'
+import { TelemetryRepository } from 'core/repositories/telemetry'
+import AuthService, { TOKEN_AUDIENCE } from 'core/services/auth/auth-service'
+import { RequestHandler, Router } from 'express'
+import Joi from 'joi'
+
+import { CustomRouter } from './customRouter'
+import { checkTokenHeader } from './util'
+
+export class TelemetryRouter extends CustomRouter {
+  private checkTokenHeader: RequestHandler
+
+  constructor(
+    private logger: sdk.Logger,
+    private authService: AuthService,
+    private telemetryRepo: TelemetryRepository
+  ) {
+    super('Telemetry', logger, Router({ mergeParams: true }))
+    this.checkTokenHeader = checkTokenHeader(this.authService, TOKEN_AUDIENCE)
+    this.setupRoutes()
+  }
+
+  private setupRoutes() {
+    // TODO get server header
+    this.router.post(
+      '/feedback',
+      this.checkTokenHeader,
+      this.asyncMiddleware(async (req, res) => {
+        const { success, events } = req.body
+
+        const schema = Joi.object({
+          success: Joi.boolean().required(),
+          events: Joi.array()
+            .items(Joi.string())
+            .required()
+        })
+
+        if (schema.validate({ success, events })) {
+          res.sendStatus(400)
+          return
+        }
```

Comment:
Your joi object is not correct, you are missing the keys property. Did you test it ? Also, for validation you can use it this way instead, it will throw if the format is invalid and give a detailed error

```js
 this.asyncMiddleware(async (req, res) => {
        const { success, events } = req.body

        await Joi.validate(
          req.body,
          Joi.object().keys({
            success: Joi.boolean().required(),
            events: Joi.array()
              .items(Joi.string())
              .required()
          })
        )
        
        success
          ? await this.telemetryRepo.removeMany(events)
          : await this.telemetryRepo.updateAvailability(events, true)
        res.sendStatus(200)
      })
```
EX:
![image](https://user-images.githubusercontent.com/42552874/89463352-ccab2800-d73c-11ea-8949-660062a84dbd.png)

Reply:
I think that my "yarn build" had a problem! It happened once before where building botpress would not register and the changes were not there. I thought that it worked.

Reply:
Anyway... I'll try these changes and commit these changes if they work

Reply:
Fixed in 69e126d!